### PR TITLE
Update Ubuntu 20.04 CIS Benchmark

### PR DESCRIPTION
**Pull request by shanesiebken**
#### Description:

- Add new rules to existing (slightly outdated) ubuntu 20.04 CIS benchmark

#### Rationale:

- This benchmark is pretty old. It's missing references to a bevy of new rules that bring it much closer to full coverage.

### CODE REVIEW

File: linux_os/guide/services/mail/service_postfix_enabled/rule.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: alinux2,alinux3,ol7,ol8,ol9,rhel7,rhel8,rhel9,sle12,sle15,ubuntu2204
+prodtype: alinux2,alinux3,ol7,ol8,ol9,rhel7,rhel8,rhel9,sle12,sle15,ubuntu2004,ubuntu2204
```
[the remark this answers is not in the document]
Comment (shanesiebken) [reply linux_os/guide/services/mail/service_postfix_enabled/rule.yml:3]: This oval definition is referenced by rule `postfix_network_listening_disabled`, so without this rule definition containing the ubuntu2004 prodtype that rule fails to compile into the datastream.

[the remark this answers is not in the document]
Comment (dodys) [reply linux_os/guide/services/mail/service_postfix_enabled/rule.yml:3]: oh right, thanks for confirming it!

**Review by dodys — APPROVED** [review]
lgtm

File: linux_os/guide/services/mail/service_postfix_enabled/rule.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shanesiebken) [reply linux_os/guide/services/mail/service_postfix_enabled/rule.yml:3]: No problem at all! It took me some digging to figure out. Thank you for the review!

File: linux_os/guide/services/ssh/sshd_strong_kex.var [hunk 2]
```diff
@@ -14,3 +14,4 @@ options:
     default: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
     cis_sle12: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
     cis_sle15: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
+    cis_ubuntu2004: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
```
Comment (marcusburghardt) [inline linux_os/guide/services/ssh/sshd_strong_kex.var:17]: Could you include an empty line at the end, please?

File: products/ubuntu2004/profiles/cis_level1_server.profile [hunk 3]
```diff
@@ -880,19 +888,19 @@ selections:
...
-    # Needs rule: no_duplicate_user_names
+    - account_unique_name
 
     ### 6.2.16 Ensure no duplicate group names exist (Automated)
-    # Needs rule: no_duplicate_group_names
+    - group_unique_name
 
     ### 6.2.17 Ensure shadow group is empty (Automated)
-    # Needs rule: ensure_shadow_group_empty
+    - ensure_shadow_group_empty
```
Comment (marcusburghardt) [inline products/ubuntu2004/profiles/cis_level1_server.profile:906]: Could you include an empty line at the end, please?

File: linux_os/guide/services/ssh/sshd_strong_kex.var (hunk 2 above)
Comment (shanesiebken) [reply linux_os/guide/services/ssh/sshd_strong_kex.var]: done

File: products/ubuntu2004/profiles/cis_level1_server.profile (hunk 3 above)
Comment (shanesiebken) [reply products/ubuntu2004/profiles/cis_level1_server.profile]: done

**Review by marcusburghardt — APPROVED** [review]

**Pull request merged**